### PR TITLE
[DOC-64] sync disasters, countries and disaster_types

### PR DESCRIPTION
**Pull request by lazysoundsystem**
Trying to update the disasters, I needed to first update the countries (the fields were all missing, including iso3), disaster types (disaster_type_code was missing) and disaster_status terms (none created).

As with https://github.com/UN-OCHA/docstore-site/pull/159 - this is a bit messy and would benefit from `drush_shift()`. It takes an argument so it can be run regularly as a cron job to just update recent disasters, but old ones do get updated and need to be synced too.

There's an issue with the provider_uuid (I think) that I haven't solved yet, but I'm not sure how important it is.

### CODE REVIEW

File: html/modules/custom/docstore/syncs/docstore_disasters.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+
+  // Check for more data.
+  if (isset($data->links) && isset($data->links->next->href)) {
+    print $data->links->next->href;
+    syncDisasters($date, $data->links->next->href);
+  }
+}
+
+// drush_shift() isn't working here. It would be an improvement.
```
Comment (attiks) [inline html/modules/custom/docstore/syncs/docstore_disasters.php:211]: I think you need a real drush command

File: html/modules/custom/docstore/syncs/docstore_disasters.php [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+ * Using the RW ids for the id field. Not sure if this needs clarifying.
+ *
+ * @todo this gives some errors connected to revisionUser uuid, which isn't
+ * defined:
+ * `[warning] array_flip(): Can only flip STRING and INTEGER values!
+ * EntityStorageBase.php:266`
+ */
+function syncDisasters($date = '', $url = '') {
+
+  // @todo is this right?
```
Comment (attiks) [inline html/modules/custom/docstore/syncs/docstore_disasters.php:20]: yes

File: html/modules/custom/docstore/syncs/docstore_disasters.php [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      $item['provider_uuid'][] = [
+        'target_uuid' => $provider->uuid(),
+      ];
+
+      // Set creation time.
+      $item['created'][] = [
+        'value' => time(),
+      ];
+
+      // @todo is this right?
```
Comment (attiks) [inline html/modules/custom/docstore/syncs/docstore_disasters.php:66]: Yes, comment about hid is wrong

**Review by attiks — APPROVED** [review]

**Comment by lazysoundsystem** [comment]
Redone as drush commands, which keeps it a tidier and more usable. The commands to update countries and disaster_types should be run before updating disasters.

@attiks apart from moving to drush commands, there are only minimal changes from the version you already checked, but probably worth another quick review.

File: html/modules/custom/docstore/src/Commands/DocstoreCommands.php [hunk 4]
```diff
@@ -137,6 +148,530 @@ public function __construct(
...
+            $value = $row->fields->{$name};
+          }
+
+          $term->set($field_name, $value);
+        }
+      }
+
+      $violations = $term->validate();
+      if (count($violations) > 0) {
+        print($violations->get(0)->getMessage());
```
Comment (attiks) [inline html/modules/custom/docstore/src/Commands/DocstoreCommands.php:454]: `drush_log` ?

**Review by attiks — APPROVED** [review]
Looking good

**Pull request merged**